### PR TITLE
Add Grades feature

### PR DESCRIPTION
**Pull request by Justin-Fernbaugh**
# Pull Request Template

## Description
I've added a Grades table that populates or updates the grades of a user after they respond to a question(Responses.js). Moreover, these changes involve using points that are added to questions using the 'weights' attribute of a question. The 'weights' of a question can be added or if they're not present will be auto-populated with a default value of '1' for each.

Important: before testing this branch the databases have to be dropped, re-added, and then migrated/seeded. The first four steps must be performed in MySQL while steps 5-8 are performed in the dir server/
1. DROP DATABASE myclassroom_development;
2. CREATE DATABASE myclassroom_development;
3. DROP DATABASE myclassroom_test;
4. CREATE DATABASE myclassroom_test;
5. npx sequelize-cli db:migrate --env test
6. npx sequelize-cli db:migrate --env dev
7. npx sequelize-cli db:seed:all --env test
8. npx sequelize-cli db:seed:all --env dev

Fixes # (issue)

## Checklist:
Before you submit your Pull Request, please make sure you have completed the following tasks:
- [X] I have performed a self-review of my own code.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [X] I have made corresponding changes to the documentation.
- [X] My changes generate no new warnings.
- [X] I have added tests that prove my fix is effective or that my feature works.
- [X] New and existing unit tests pass locally with my changes.
- [X] Any dependent changes have been merged and published in downstream modules.
- [X] I have tagged my PR with the appropriate label(s).

### CODE REVIEW

File: server/app/api/users.js [hunk 1]
```diff
@@ -1,422 +1,445 @@
...
+			}
+		}
+	} else {
+		res.status(400).send({
+			error: `Missing fields required to authenticate user: ${serializeStringArray(
+				missingFields
+			)}`,
+		});
+	}
+});
```
[the remark this answers is not in the document]
Comment (Justin-Fernbaugh) [reply server/app/api/users.js:186]: @karinocheretny Did you happen to get this same alert on your user changes? I assume its from when I merged dev into this branch

**Review by nilsstreedain — CHANGES_REQUESTED** [review]
Looks like this PR has the same issue that @Durbin-Elijah's did. If you remember, awhile back, we had an unknown contributor who made a PR to `main` that got accepted, causing divergent history between `main`/`dev` because all changes on `main` were intended to come directly from `dev`, nowhere else. It appears the branch submitted in this PR was pulled from `main` instead of `dev` causing it to try to merge those non-existent commits to `dev`. Would it be possible to go through and remove any commits that are not apart of this change so once we are ready for a new version, I can force override `main`? (Only force overriding because our branches should be built from `dev`, not `main`.)

**Comment by Justin-Fernbaugh** [comment]
> Looks like this PR has the same issue that @Durbin-Elijah's did. If you remember, awhile back, we had an unknown contributor who made a PR to `main` that got accepted, causing divergent history between `main`/`dev` because all changes on `main` were intended to come directly from `dev`, nowhere else. It appears the branch submitted in this PR was pulled from `main` instead of `dev` causing it to try to merge those non-existent commits to `dev`. Would it be possible to go through and remove any commits that are not apart of this change so once we are ready for a new version, I can force override `main`? (Only force overriding because our branches should be built from `dev`, not `main`.)

Yeah, I should be able to do this, would you be able to point me to which commit this starts so I can remove the changes?

**Comment by nilsstreedain** [comment]
> > Looks like this PR has the same issue that @Durbin-Elijah's did. If you remember, awhile back, we had an unknown contributor who made a PR to `main` that got accepted, causing divergent history between `main`/`dev` because all changes on `main` were intended to come directly from `dev`, nowhere else. It appears the branch submitted in this PR was pulled from `main` instead of `dev` causing it to try to merge those non-existent commits to `dev`. Would it be possible to go through and remove any commits that are not apart of this change so once we are ready for a new version, I can force override `main`? (Only force overriding because our branches should be built from `dev`, not `main`.)
> 
> Yeah, I should be able to do this, would you be able to point me to which commit this starts so I can remove the changes?

Looks like just commits c6524de, 86d4753, and 5ed0d93 need to be removed, preferably with a rebase. If anything is left committing to `/client`, that should also probably be removed.

**Comment by Justin-Fernbaugh** [comment]
Merged with changes from dev and dropped any changes from main. Ready for a second look

File: server/.env.example
Comment (nilsstreedain) [inline server/.env.example:1]: Merge conflict remnants

File: .gitignore [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 # dependencies
 /node_modules
+/client/node_modules
```
Comment (nilsstreedain) [inline .gitignore:5]: Duplicate line

**Review by nilsstreedain — CHANGES_REQUESTED** [review]
Generally looks great, see comments for 2 minor changes I found.

File: .gitignore (hunk 2 above)
Comment (Justin-Fernbaugh) [reply .gitignore]: Pushed

File: server/.env.example
Comment (Justin-Fernbaugh) [reply server/.env.example:1]: Pushed

**Review by nilsstreedain — DISMISSED** [review]
Looks great!

**Review by nilsstreedain — APPROVED** [review]
Looks great, nice job!

**Pull request merged**